File: languages/python/bloock/entity/identity/issuer_state_receipt.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto

class IssuerStateReceipt:
    """
    Represents a receipt for the issuer's state.
    """
    def __init__(self, txHash: str) -> None:
        """
        Constructs an IssuerStateReceipt object with the specified parameters.
        :type txHash: object
        :rtype: object
        """
        self.txHash = txHash

    @staticmethod
    def from_proto(c: proto.IssuerStateReceipt) -> IssuerStateReceipt:
        return IssuerStateReceipt(
            txHash=c.tx_hash
        )

    def to_proto(self) -> proto.IssuerStateReceipt:
        return proto.IssuerStateReceipt(
            tx_hash=self.txHash
        )

File: languages/php/Bloock/Entity/Identity/IssuerStateReceipt.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents a receipt for the issuer's state.
 */
class IssuerStateReceipt
{
    private string $txHash;

    /**
     * Constructs an IssuerStateReceipt object with the specified parameters.
     * @param string $txHash
     */
    public function __construct(string $txHash)
    {
        $this->txHash = $txHash;
    }

    public static function fromProto(\Bloock\IssuerStateReceipt $res): IssuerStateReceipt
    {
        return new IssuerStateReceipt($res->getTxHash());
    }

    /**
     * Gets the transaction hash of the issuer state receipt.
     * @return string
     */
    public function getTxHash(): string
    {
        return $this->txHash;
    }

    public function toProto(): \Bloock\IssuerStateReceipt
    {
        $p = new \Bloock\IssuerStateReceipt();
        $p->setTxHash($this->txHash);
        return $p;
    }
}